src/components/checkBoxSelection/checkBoxSelection.js

```diff
@@ -1,47 +1,112 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import './checkBoxSelection.scss';
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import "./checkBoxSelection.scss";
+import { connect } from "react-redux";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class CheckBoxSelection extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: false
-        };
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected
+    };
+  }
+
+  convertSvgToDataUrl = file => {
+    let reader = new FileReader();
+    reader.onload = result => {
+      this.props.svgSettingList.push({
+        name: file.name,
+        dataurl: result.target.result
+      });
+    };
+    if (file) {
+      reader.readAsDataURL(file);
     }
+  };
 
-    componentWillMount() {
-	 this.setState({ isClicked: this.props.isSelected }); 
+  changeColourAndTick = () => {
+    let isFound = false;
+    let index = 0;
+    this.setState({ isClicked: !this.state.isClicked });
+    for (var key in this.props.svgSettingList) {
+      if (
+        this.props.svgSettingList[key].name === this.props.file.name
+      ) {
+        isFound = true;
+        index = key;
+      }
     }
```

Comment:
use array.find here

Reply:
The other approach would be like myArray = [{'id':'73','foo':'bar'},{'id':'45','foo':'bar'}, etc.] , which I would scan like myArray.find(x => x.id === '45') and seems to be the same as above. This is the same approach and the one I provided gives me lot of flexibility.

Reply:
Since I am also looking for theitem and its index I have decided to use break so it does not loop even after it have found the item.

Reply:
array.findIndex

Reply:
for is too inefficent, it might make it easier for you to code in the future, but the cost is efficiency. code for now with an intention for the future. but never compromise what is better for right now for what might possibly happen in the future. ie, never comprimise what is know for what is possible or a maybe

Reply:
Done.

Reply:
I have decided to use findIndex and apparently I did not need the item itself but just its index.

Reply:
im glad, did you also read the documentation for "for" vs find and see that find stops as soon as it finds a result where as for runs every time no matter what